terraform/Makefile

```diff
@@ -34,15 +32,15 @@ $(GO_BUILD_TARGETS): go-build.%: bin/$(TARGET_OS_ARCH)/terraform-provider-%.zip
 $(TERRAFORM_PROVIDER_TARGETS): bin/$(TARGET_OS_ARCH)/terraform-provider-%.zip: providers/%/go.mod
 	cd providers/$*; \
 	if [ -f main.go ]; then path="."; else path=./vendor/`grep _ tools.go|awk '{ print $$2 }'|sed 's|"||g'`; fi; \
-	go build -gcflags $(GCFLAGS) -ldflags $(LDFLAGS) -o ../../bin/$(TARGET_OS_ARCH)/terraform-provider-$* "$$path" && \
+	go build -gcflags $(GCFLAGS) -ldflags "$(LDFLAGS) -X installer.version=$$(git hash-object go.mod)" -o ../../bin/$(TARGET_OS_ARCH)/terraform-provider-$* "$$path" && \
```

Comment:
According to https://github.com/golang/go/wiki/GcToolchainTricks#including-build-information-in-the-executable "The string variable must exist, ..." but "There is no warning for using the wrong name".
It looks like we're relying on the fact that the ldflags are stored in the binary rather than anything the linker is actually doing with them? It would be good to mention that , at least in the commit message if a comment is too unwieldy.
A more descriptive name might be helpful for some future maintainer too.

Reply:
What name would you suggest?

Reply:
Maybe something like `main.builtGoModHash`?